system-tests-new/incremental/test_incremental.py

```diff
@@ -224,3 +228,158 @@ def assert_one_artefact(self, pb_keys, prebuild_groups, prebuild_folder,
             pb_fpath = next(iter(pb_group))
             assert clean_timestamps[prebuild_folder / pb_fpath] == rebuild_timestamps[prebuild_folder / pb_fpath]
             assert clean_hashes[prebuild_folder / pb_fpath] == rebuild_hashes[prebuild_folder / pb_fpath]
+
+
+class TestCleanupPrebuilds(object):
+    # Test cleanup of the incremental build artefacts
+
+    def create_artefacts(self, folder):
```

Comment:
You could make this a fixture itself:

```python
@pytest.fixture
def test_tree(tmp_path: Path):
    ...
    return tmp_path
...
def test_prunt_oldest_15(self, test_tree: Path)
```

Reply:
I tried at the time but didn't find a way. For `test_prune_oldest_15`, the `tmp_path` fixture is passed through to `_prune_oldest` which creates and returns the config and artefacts. I don't understand if the above snippet can help with that?